bionic/flow.py

```diff
@@ -1655,31 +1657,87 @@ def core__persistent_cache(
 
     @builder
     @decorators.immediate
-    def core__process_executor(core__parallel_processing__enabled):
+    def core__process_manager(core__parallel_processing__enabled):
         if not core__parallel_processing__enabled:
             return None
 
-        # Loky uses cloudpickle by default. We should investigate further what would
-        # it take to make is use pickle and wrap the functions that are non-picklable
-        # using `loky.wrap_non_picklable_objects`.
-        loky = import_optional_dependency("loky", purpose="parallel processing")
-        # TODO: Use a config / cpu cores instead of using a hardcoded value.
-        # Same applies to the test executor.
-        return loky.get_reusable_executor(max_workers=2)
+        return Manager()
 
     @builder
     @decorators.immediate
-    def core__process_manager(core__parallel_processing__enabled):
+    def core__logging_queue(core__parallel_processing__enabled, core__process_manager):
+        if not core__parallel_processing__enabled:
+            return None
+
+        if core__process_manager is None:
+            raise AssertionError("core__process_manager is None, but needs a value")
+
+        return core__process_manager.Queue(-1)
+
+    @builder
+    @decorators.immediate
+    def core__logging_receiver(core__parallel_processing__enabled, core__logging_queue):
+        if not core__parallel_processing__enabled:
+            return None
+
+        if core__logging_queue is None:
+            raise AssertionError("core__logging_queue is None, but needs a value")
+
+        logging_receiver = LoggingReceiver(core__logging_queue)
```

Comment:
It does start a new receiver thread every time we instantiate a flow. If a user creates two flows out of the same builder and using both flows, I think we should have 2 separate receiver instances for each flow (instead of sharing them). I'm creating a different instance of the queue for each flow too. In fact, we do the same with executor and manager too, create a new one for each flow. 

We can change it and instantiate them creating a builder but then we lose the ability to override them since they aren't bionic entities anymore. Wdyt?

I think moving all of this in a single class makes sense (including the manager).